File: app/models/db/script_run.rb
# frozen_string_literal: true

# Represents a single run of a script
# holds its output logs and state
class ::DB::ScriptRun < ::ApplicationRecord
  belongs_to :script
  belongs_to :initiator, class_name: '::DB::User'

  default_scope { order(id: :desc) }

  enum state: { waiting: 0, running: 1, executed: 2, failed: 3, timed_out: 4, connection_failed: 5 }
end

File: frontend/src/types/script-run.ts
export type ScriptRunState = 'waiting' | 'running' | 'executed' | 'failed' | 'timed_out' | 'connection_failed'

export interface ScriptRun {
  id: number
  scriptId: number
  input: string
  output: string
  initiatorId: number
  state: ScriptRunState
  triggeredAt: string
  delay: number
  executedAt: string | null
  scriptName: string
  url: string
}
